src/olympia/devhub/views.py

```diff
@@ -1940,6 +1940,8 @@ def onboarding_subscription_success(request, addon_id, addon):
             payment_completed_at=datetime.datetime.now(),
         )
         log.info('PromotedSubscription %s has been completed.', sub.pk)
+        if not sub.addon_already_promoted:
+            sub.promoted_addon.approve_for_addon()
```

Comment:
Not sure how/if we should solve this but paid subscriptions on -dev won't have the add-ons auto-approved. The success page will say "add-on will be re-signed" but that won't happen. (I am sure QA will verify that)

WDYT?

Reply:
On Element, we decided to truncate the promoted subscription in -dev. It won't be possible to trigger such a scenario in stage/prod.